### PR TITLE
096 rc1 prep

### PR DESCRIPTION
**Pull request by newville**
This prepares for 0.9.6rc1:

It's sort of a huge set of changes, but looks much bigger than it actually is.  The changes are dominated by moving text from RST docs to code docstrings  so that the docstrings are now used for all methods and classes in the RST docs.  

In addition:  differential_evolution is now expected to be available from scipy.optimize, which allowed for a little bit of cleanup in how this was called.   I added a very small example using diffev, if only to demonstrate that it works.

The docs may still need improvements, of course.

### CODE REVIEW

File: examples/example_diffev.py [hunk 1]
```diff
@@ -0,0 +1,60 @@
+#!/usr/bin/env python
+"""
+Example using the Student's t log-likelihood
+for robust fitting of data with outliers.
```
Comment (andyfaff) [inline examples/example_diffev.py]: The `resid` function looks like a normal residual. I can't see why this is using Student's t log-likelihood.

File: lmfit/minimizer.py [hunk 2]
```diff
@@ -33,19 +34,15 @@
...
 #    diff_evolution        0.15
 #    least_squares         0.17
 
-# differential_evolution is only present in scipy >= 0.15
-try:
-    from scipy.optimize import differential_evolution as scipy_diffev
-except ImportError:
-    from ._differentialevolution import differential_evolution as scipy_diffev
 
+from scipy.optimize import differential_evolution as scipy_diffev
```
Comment (andyfaff) [inline lmfit/minimizer.py]: why not `from scipy.optimize import differential_evolution`? Is it necessary to bind it to the `scipy_diffev` name?

File: lmfit/minimizer.py [hunk 3]
```diff
@@ -921,8 +920,8 @@ def emcee(self, params=None, steps=1000, nwalkers=100, burn=0, thin=1,
         `is_weighted is False` then the data uncertainty, `s_n`, will be
         treated as a nuisance parameter and will be marginalized out. This is
         achieved by employing a strictly positive uncertainty
-        (homoscedasticity) for each data point, :math:`s_n = \exp(\_\_lnsigma)`.
-        `__lnsigma` will be present in `MinimizerResult.params`, as well as
+        (homoscedasticity) for each data point, :math:`s_n = \exp(\_lnsigma)`.
+        `_lnsigma` will be present in `MinimizerResult.params`, as well as
```
Comment (andyfaff) [inline lmfit/minimizer.py]: Are you sure it's one underscore?

File: lmfit/minimizer.py [hunk 4]
```diff
@@ -703,21 +701,22 @@ def scalar_minimize(self, method='Nelder-Mead', params=None, **kws):
...
+            internal_bounds = [(-np.pi / 2., np.pi / 2.)] * len(vars)
+            kwargs = dict(args=(), strategy='best1bin', maxiter=None,
+                          popsize=15, tol=0.01, mutation=(0.5, 1),
+                          recombination=0.7, seed=None, callback=None,
+                          disp=False, polish=True, init='latinhypercube')
+
+            for k, v in fmin_kws.items():
+                if k in kwargs:
+                    kwargs[k] = v
+            ret = scipy_diffev(self.penalty, internal_bounds, **kwargs)
```
Comment (andyfaff) [inline lmfit/minimizer.py]: `differential_evolution`, not `scipy_diffev`?

File: lmfit/parameter.py [hunk 5]
```diff
@@ -295,31 +325,27 @@ def add(self, name, value=None, vary=True, min=-inf, max=inf, expr=None,
...
-        # add a sequence of Parameter
-        f = Parameter('name5', val5)
-        g = Parameter('name6', val6)
-        p.add_many(f, g)
+            name. The order in each tuple must be `(name, value, vary, min, max, expr, brute_step)`
 
+        Examples
+        --------
+        >>>  params = Parameters()
+        # add with tuples: (NAME VALUE VARY MIN  MAX  EXPR  BRUTE_STEP)
```
Comment (andyfaff) [inline lmfit/parameter.py]: Do doctests get run when the test suite runs?

File: examples/example_diffev.py (hunk 1 above)
Comment (newville) [inline examples/example_diffev.py]: thanks, will fix

File: lmfit/minimizer.py (hunk 2 above)
Comment (newville) [inline lmfit/minimizer.py]: good point, will change

File: lmfit/minimizer.py (hunk 4 above)
Comment (newville) [inline lmfit/minimizer.py]: will change

File: lmfit/minimizer.py (hunk 3 above)
Comment (newville) [inline lmfit/minimizer.py]: no, I'm not sure.  Will check.

File: lmfit/parameter.py (hunk 5 above)
Comment (newville) [inline lmfit/parameter.py]: I think doctests are not run.

**Comment by newville** [comment]
@andyfaff Thanks!  Yes, I knew this would touch many files in complex ways, so basically waited until almost all other PRs and issues had quieted down.  I think this puts any other PRs on hold and probably in need of rebasing.

I'll try to incorporate all your comments within a day.

**Comment by reneeotten** [comment]
@newville thanks for making this big improvements on docstrings and documentation! There are some typos left here and there, but I would be happy to go over the docstrings/documentation once more and try to correct these. It's probably easier to merge this first so that the few left-over PRs can be rebased and finished up as well, while I take a look at it during this week?

**Comment by newville** [comment]
@reneeotten Thanks!  It is sort of a huge set of changes, and I am sure there are many doc typos. Having another set of eyes go over this would be very much appreciated.   I'll merge this now.

**Pull request merged**